OpenSim/Common/XMLDocument.cpp

```diff
@@ -404,23 +404,60 @@ bool XMLDocument::isElementEqual(SimTK::Xml::Element& elt1, SimTK::Xml::Element&
 /*
  * Helper function to add connector to the xmlElement passed in
  */
-void XMLDocument::addConnector(SimTK::Xml::Element& element, const std::string& connectorTag, const std::string& connectorName, const std::string& connectorValue)
+void XMLDocument::addConnector(SimTK::Xml::Element& element,
+    const std::string& connectorTag, const std::string& connectorName, 
+    const std::string& connectorValue)
 {
     SimTK::Xml::element_iterator  connectors_node =  element.element_begin("connectors");
-    SimTK::String debug;
+    //SimTK::String debug; //Only used for debugging
     if (connectors_node == element.element_end()){
         SimTK::Xml::Element connectorsElement("connectors");
         element.insertNodeBefore(element.element_begin(), connectorsElement);
         connectors_node =  element.element_begin("connectors");
     }
-    // Here we're guaranteed connectors node exist, add individual connector
+    // Here we're guaranteed connectors node exists, add individual connector
     SimTK::Xml::Element newConnectorElement(connectorTag);
     newConnectorElement.setAttributeValue("name", connectorName);
-    newConnectorElement.writeToString(debug);
+    //newConnectorElement.writeToString(debug);
+
     SimTK::Xml::Element connecteeElement("connectee_name");
     connecteeElement.insertNodeAfter(connecteeElement.element_end(), SimTK::Xml::Text(connectorValue));
     // Insert text under newConnectorElement
     newConnectorElement.insertNodeAfter(newConnectorElement.element_end(), connecteeElement);
     connectors_node->insertNodeAfter(connectors_node->element_end(), newConnectorElement);
-    connectors_node->writeToString(debug);
+    //connectors_node->writeToString(debug);
+}
+
+void XMLDocument::addPhysicalOffsetFrame(SimTK::Xml::Element& element,
+    const std::string& frameName,
+    const std::string& parentFrameName, 
+    const SimTK::Vec3& location, const SimTK::Vec3& orientation)
+{
+    SimTK::Xml::element_iterator  frames_node = element.element_begin("frames");
+    //SimTK::String debug; //Only used for debugging
+    
+    if (frames_node == element.element_end()) {
+        SimTK::Xml::Element framesElement("frames");
+        element.insertNodeBefore(element.element_begin(), framesElement);
+        frames_node = element.element_begin("frames");
+    }
+    // Here we're guaranteed frames node exists, add individual frame
+    SimTK::Xml::Element newFrameElement("PhysicalOffsetFrame");
+    newFrameElement.setAttributeValue("name", frameName);
```

Comment:
This looks like it would allow multiple frames of the same name in the "frames" element. Is that OK?